src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -30,7 +32,7 @@ public Task Complete(IContainSagaData saga, ContextBag context)
                 {
                     continue;
                 }
-                if (!Equals(prop.GetValue(entity.SagaEntity, null), propertyValue))
+                if (prop.GetValue(entity.SagaEntity).ToString() != propertyValue.ToString())
```

Comment:
Had to revert this, since the InMemory persister is cloning the value we can't use Equals since that doesn't  work for `Guid`. Given that we now check the types uses to correlate with I'd say `.ToString` is ok?